### ISSUE TITLE
"Bad private key" from an outside private key

### ISSUE THREAD

**Issue by tnguyen42** [bitchan/eccrypto#48]
Hi,

I've been trying to use the getPublic function on a random private key just to try it out, but I keep getting the error "Bad private key".

Here's my code:
```
const privateKeyTest =
			"0x8da4ef21b864d2cc526dbdb2a120bd2874c36c9d0a1fb7f8c63d7f7a8b41de8f";

let privateKeyTable = [];
		for (let i = 2; i < privateKeyTest.length; i = i + 2) {
			const newElement = parseInt(privateKeyTest.slice(i, i + 2), 16);
			privateKeyTable.push(newElement);
		}

const TypedTable = Uint8Array.from(privateKeyTable);

		console.log(TypedTable);

const publicKey = eccrypto.getPublic(TypedTable);
		console.log(publicKey);

```

I've been able to use getPublic after eccrypto.generatePrivate(), and TypedTable is indeed a Uint8Array. 
I also tried with a generated private key in hard code, such as this:

```
		const TypedTable = new Uint8Array([
			235,
			78,
			133,
			244,
			5,
			163,
			18,
			32,
			218,
			162,
			173,
			132,
			214,
			16,
			57,
			100,
			122,
			48,
			167,
			30,
			203,
			69,
			52,
			48,
			30,
			99,
			20,
			32,
			249,
			210,
			140,
			48
		]);

const publicKey = eccrypto.getPublic(TypedTable);
```
And it still does not work. Is there a reason why?

Thanks!

**Comment by JBaczuk** [comment]
Are you using node.js? If so, you need to use Buffer instead of Uint8Array. If not, are you using a transpiler like webpack?

**Comment by JBaczuk** [comment]
closing for now, if you still need help, reopen

**Comment by xpepermint** [comment]
@JBaczuk it's not clear how to transform the private key in the browser that would be accepted by the `decrypt` method. Can you help?

**Comment by xpepermint** [comment]
I can confirm that the example from the README works in NodeJS but not in the browser. Try running the code below in the browser yourself:

```js
let eccryptoJS = window.eccryptoJS;
let publicKey = hexToString('...');
let privateKey = hexToString('...');
keyPair = { publicKey, privateKey};

const str = 'test message to encrypt';
const msg = eccryptoJS.utf8ToBuffer(str);

const encrypted = await eccryptoJS.encrypt(keyPair.publicKey, msg);
console.log('encrypted:', encrypted);
const decrypted = await eccryptoJS.decrypt(keyPair.privateKey, encrypted);
console.log('decrypted:', decrypted);
```
```js
function hexToString(hexString) {
	return new Uint8Array(hexString.match(/.{1,2}/g).map(byte => parseInt(byte, 16)));
}
```

**Comment by fulldecent** [comment]
I have added a cash commitment to fix this issue at #72

**Comment by ChaituVR** [comment]
The problem here is with 
https://github.com/bitchan/eccrypto/blob/ee2518544ca36849e4618907dea3c8e3a34a00b0/browser.js#L21
as mentioned in https://github.com/bitchan/eccrypto/issues/48#issuecomment-565681777

If you are not using browserify we cant have a `Buffer`, so the quick fix for you here is to use https://github.com/feross/buffer
instead of `Uint8Array`

and for CDN we can use unpkg's eccrypto-js which supports file integrity 
https://unpkg.com/eccrypto-js@5.3.0/dist/umd/index.min.js

and here is the codepen for working version
https://codepen.io/ChaituVR/pen/GRjKEMZ

below is HTML that is working
```html
<head>
<title>LET's ENCRYPT</title>
<script src="https://bundle.run/buffer@6.0.3"></script>
<script>
    window.Buffer = buffer.Buffer
</script>
<script src="https://unpkg.com/eccrypto-js@5.3.0/dist/umd/index.min.js"></script>
<script>
    function hexToString(hexString) {
        return new Buffer.from(hexString.match(/.{1,2}/g).map(byte => parseInt(byte, 16)));
    }
    (async function(){
      let eccryptoJS = window.eccryptoJS;
      let privateKey = hexToString("8da4ef21b864d2cc526dbdb2a120bd2874c36c9d0a1fb7f8c63d7f7a8b41de8f");
      let publicKey = eccryptoJS.getPublic(privateKey);
      keyPair = { publicKey, privateKey};
    
      const str = 'test message to encrypt';
      const msg = eccryptoJS.utf8ToBuffer(str);
    
      const encrypted = await eccryptoJS.encrypt(keyPair.publicKey, msg);
      console.log('encrypted:', encrypted);
      const decrypted = await eccryptoJS.decrypt(keyPair.privateKey, encrypted);
      console.log('decrypted:', decrypted);
      let decryptedstring = new TextDecoder("utf-8").decode(decrypted);
      console.log(decryptedstring)
    })()</script>
</head>
<body>
  OPEN CONSOLE
</body>
```


PS: https://bundle.run/buffer@6.0.3 this CDN is recommended in https://github.com/feross/buffer
